Orange/widgets/utils/state_summary.py

```diff
@@ -55,10 +55,50 @@ def _plural(number):
         targets = format_variables_string(data.domain.class_vars)
         metas = format_variables_string(data.domain.metas)
 
+        features_missing = missing_values(data.has_missing_attribute()
+                                          and data.get_nan_frequency_attribute())
         n_features = len(data.domain.variables) + len(data.domain.metas)
         details = \
             f'{len(data)} instance{_plural(len(data))}, ' \
             f'{n_features} variable{_plural(n_features)}\n' \
-            f'Features: {features}\nTarget: {targets}\nMetas: {metas}'
-
+            f'Features: {features} {features_missing}\n' \
+            f'Target: {targets}\nMetas: {metas}'
     return details
+
+
+def missing_values(value):
+    if value:
+        return f'({value*100:.1f}% missing values)'
+    else:
+        return '(No missing values)'
+
+
+def format_multiple_summaries(data_list, type_io='input'):
+    """
+    A function that forms the entire descriptive part of the input/output
+    summary for widgets that have more than one input/output.
+
+    :param data_list: A list of tuples for each input/output dataset where the
+    first element of the tuple is the name of the dataset (can be omitted)
+    and the second is the dataset
+    :type data_list: list(tuple(str, Orange.data.Table))
+    :param type_io: A string that indicates weather the input or output data
+    is being formatted
+    :type type_io: str
+
+    :return A formatted summary
+    :rtype str
+    """
+
+    def new_line(text):
+        return text.replace('\n', '<br>')
+
+    full_details = []
+    for (name, data) in data_list:
+        if data:
+            details = new_line(format_summary_details(data))
+        else:
+            details = 'No data on input' if type_io == 'input' else \
```

Comment:
The IF statement could be shortened:
`details = f'No data on {type_io}.'`

Plus, I'd prefer a dot at the end.